spec/features/user/note/search_spec.rb

```diff
@@ -0,0 +1,28 @@
+require "rails_helper"
+
+feature "Search notes" do
+  before do
+    create(:note, title: "alpha beta", content: "gamma delta")
+    create(:note, title: "beta epsilon", content: "theta delta")
+    create(:note, title: "one two", content: "three four")
+  end
+
+  scenario "User searches notes by title" do
+    visit notes_path
+
+    fill_in :search_form_query, with: "beta"
+    click_button "Search"
+
+    # first note found
```

Comment:
Usually, we do not write comments in the code, except in rare cases with very specific logic that is difficult to understand.